.travis.yml

```diff
@@ -36,12 +37,28 @@ jobs:
       - |
           cd tests/integration/metrics
           PYTHONPATH="$TRAVIS_BUILD_DIR/src" pytest -v
-  - stage: docker-deploy
-    name: Docker Hub deploy
-    if: tag =~ /^release-.*
+  - stage: docker-build
```

Comment:
Should it be separate stage if we are duplicating so much from `e2e-deployment-test` to run single command `"./build/docker-deploy.sh"`?

Reply:
this is an ideological question. It depends. If we would like to have flow that first build artifact (docker) then test it answer is yes, but if we are good with building docker pushing it and then downloading it again in the same step it can be also there

Reply:
GitLab allows inherit setting and here is a ideal use case of that https://docs.gitlab.com/ee/ci/yaml/#merge-details but unfortunately Travis don't allow that syntax or I miss something

Reply:
Seems a bit superflous if artifact is only used for `e2e-deployment-test` stage, and we depend on it to cleanup, but I trust your judgement.